app/src/navigation/screens/DocumentDetailScreen.tsx

```diff
@@ -0,0 +1,219 @@
+  const document = route.params.document;
+
+  const { files, addFile, deleteFile, addDocument } = useDocumentStore();
+  const { selectPhotoFromCameraRoll, selectPhotoFromCamera } = useSelectPhoto(
+    DOCUMENT_IMAGE_OPTIONS
+  );
+
+  const [showAttachModal, setShowAttachModal] = React.useState<boolean>(false);
+  const [formState, setFormState] = React.useState<FormState>({});
+  const [updateDocs, setUpdateDocs] = React.useState<boolean>(false);
+
+  const selectedDocuments = React.useMemo(
+    () => files.filter((file) => document.fileIds?.includes(file.id)),
+    [files, updateDocs]
+  );
+
+  React.useEffect(() => {
+    document.fileIds = document.fileIds ? document.fileIds : [];
+    const newForm = { ...formState };
+    document.fields?.forEach((field) => {
+      newForm[field.title] = field.value ?? "";
+    });
+    setFormState(newForm);
+  }, []);
+
+  const attachFile = () => {
+    setShowAttachModal(true);
+  };
+
+  const saveDocument = () => {
+    addDocument(document);
+  };
+
+  const saveFile = async (uri: string) => {
+    const fileId = await addFile(document.type, uri);
+    document.fileIds?.push(fileId);
+    setUpdateDocs(!updateDocs);
+    saveDocument();
+  };
+
+  const _deleteFile = (fileId: string) => {
+    document.fileIds = document.fileIds?.filter((id) => id !== fileId);
+    deleteFile(fileId);
+  };
+
+  const pickPhotoFromLibrary = async () => {
+    setShowAttachModal(false);
+    const file = await selectPhotoFromCameraRoll();
+
+    if (!file.cancelled) {
+      saveFile(file.uri);
+    }
+  };
+
+  const takePhoto = async () => {
+    setShowAttachModal(false);
+    const result = await selectPhotoFromCamera();
+    if (result && !result.cancelled) {
+      saveFile(result.uri);
+    }
+  };
+
+  const uploadFile = async () => {
+    setShowAttachModal(false);
+    try {
+      const res = await DocumentPicker.getDocumentAsync({
+        type: "*/*"
+      });
+      if (res && res.type !== "cancel") {
+        saveFile(res.uri);
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const navigateToFile = (fileId: string) => {
+    navigation.navigate("ViewFile", {
+      fileId
+    });
+  };
+
+  const shouldDisableSaveButton = React.useMemo(() => {
+    const eitherOptionalOrNotEmpty = (field: any) =>
+      field.optional || field.value !== "";
+    const allRequiredFieldHaveValue = document.fields?.every(
+      eitherOptionalOrNotEmpty
+    );
+    return !allRequiredFieldHaveValue;
+  }, [document]);
+
+  const onChange = React.useCallback(
+    (input: string, field: Field) => {
+      if (!field.valueOptions || field.valueOptions.includes(input)) {
+        field.value = input;
+        setFormState((previous) => ({
+          ...previous,
+          [field.title]: input
+        }));
+      }
+    },
+    [setFormState]
+  );
+
+  return (
+    <View>
+      <Modal
+        title=""
+        show={showAttachModal}
+        onClose={() => setShowAttachModal(false)}
```

Comment:
The inverse of the `attachFile` function is done here and is not in a separate definition

Reply:
should be seperate